pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1132 @@ package listusers
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+		  type user
+			relations
+			  define blocked: [user]
+		  type folder
+			relations
+			  define viewer: [user]
+			  define blocked: blocked from viewer
+		  type document
+			relations
+			  define parent: [folder]
+			  define viewer: (viewer from parent) but not blocked from parent`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "parent", "folder:x"),
+				tuple.NewTupleKey("folder:x", "viewer", "user:will"),
+
+				tuple.NewTupleKey("folder:x", "viewer", "user:maria"),
+				tuple.NewTupleKey("user:maria", "blocked", "user:maria"),
+			},
+			expectedUsers: []string{"user:will"},
+		},
 	}
+	tests.runListUsersTestCases(t)
+}
 
-	for _, test := range tests {
+func TestListUsersWildcards(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	tests := ListUsersTests{
+		{
+			name: "direct_relationship_wildcard",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define viewer: [user:*]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:2", "viewer", "user:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+		{
+			name: "direct_relationship_wildcard_with_direct_relationships_also",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define viewer: [user:*,user]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "user:will"),
+				tuple.NewTupleKey("document:2", "viewer", "user:maria"),
+			},
+			expectedUsers: []string{"user:*", "user:will"},
+		},
+		{
+			name: "multiple_possible_wildcards_user_granularity",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+				relations
+					define member: [user]
 
+			type document
+				relations
+					define viewer: [ group:*, user:*]`,
+
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "group:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+		{
+			name: "multiple_possible_wildcards_group_granularity",
```

Comment:
Same test as the one directly above it. I vote we drop this since it's a duplicate.